api/errors/api_error.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 Nitric Technologies Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import (
+	"fmt"
+
+	"github.com/nitrictech/go-sdk/api/errors/codes"
+	"google.golang.org/grpc/status"
+)
+
+type ApiError struct {
+	code  codes.Code
+	msg   string
+	cause error
+}
+
+func (a *ApiError) Error() string {
+	if a.cause != nil {
+		// If the wrapped error is an ApiError than these should unwrap
+		return fmt.Sprintf("%s: %s: \n %s", a.code.String(), a.msg, a.cause.Error())
+	}
+
+	return fmt.Sprintf("%s: %s", a.code.String(), a.msg)
+}
+
+// FromGrpcError - translates a standard grpc error to a nitric api error
+func FromGrpcError(err error) error {
+	if s, ok := status.FromError(err); ok {
+		return &ApiError{
+			code: codes.Code(s.Code()),
+			msg:  s.Message(),
+		}
```

Comment:
what about capturing the cause:
```
		return &ApiError{
			code: codes.Code(s.Code()),
			msg:  s.Message(),
			cause: err,
		}
```

Reply:
This was a conversion, if we want to wrap the orginal error, we can probably delete this method altogether and just use `NewWithCause` instead.

thoughts?

Reply:
I think so yes.